modules/AWS/eks/nodegroup_launch_template/main.tf

```diff
@@ -0,0 +1,26 @@
+locals {
+  user_content = templatefile("${path.module}/templates/userdata.sh.tpl", {
+    account_id              = data.aws_caller_identity.current.account_id
+    aws_sts_region          = local.aws_sts_region
+    osquery_secret_name     = var.osquery_secret_name
+    osquery_secret_region   = var.osquery_secret_region
+    osquery_version        = var.osquery_version
+    env                     = var.osquery_env
+    aws_sts_arn_role        = local.aws_sts_arn_role
+    kinesis_log_producers_role_arns = var.kinesis_log_producers_role_arns
+    })
+}
+
+data "aws_caller_identity" "current" {}
+
+resource "aws_launch_template" "nodegroup" {
+  name_prefix            = "${var.cluster_name}-launch-template"
+  description            = "${var.cluster_name} Nodegroup Launch Template"
+  update_default_version = true
+  user_data = base64encode(
```

Comment:
I saw that there is `if...else...` statement in user data template to control whether enabling osquery or not. Do you think it'll be better to move the switch one level up, say if `osquery_secret_name != ""`, we don't pass in `user_data` at all?

Reply:
Yes, it's cleaner and makes more sense. Now user_data is computed in locals.